### PR TITLE
Team a1 dev

### PR DESCRIPTION
**Pull request by chowder23**

### CODE REVIEW

File: src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/HMIKeyListener.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import hu.oe.nik.szfmv.automatedcar.virtualfunctionbus.VirtualFunctionBus;
+import hu.oe.nik.szfmv.automatedcar.virtualfunctionbus.packets.HMIOutputPackets.InputPacket;
+
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+
+
+public class HMIKeyListener {
+
+    KeyProcesser Processer = new KeyProcesser();
```
Comment (AbyssPrince) [inline src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/HMIKeyListener.java:14]: Példány szintű mezőket mindig kis betűkkel nevezzétek el.

File: src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/KeyProcesser.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    public void steeringLeftPressed(){pedalPos.startSteeringLeft();}
+
+    public void steeringRightPressed(){pedalPos.startSteeringRight();}
+
+    public void steeringReleased(){pedalPos.steeringWheelReleased();}
+
+
+    private ACC accManager = new ACC();
+
+    public void turnAccSwitch(){accManager.turnAccSwitch();}
```
Comment (AbyssPrince) [inline src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/KeyProcesser.java:88]: Actually would look better as:
```java
public void turnAccSwitch() {
    accManager.turnAccSwitch();
}
```
If you `Ctrl`+`Shift`+`-` then I will be shown as one line anyways.

File: src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/KeyProcesser.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    public void breakPedalReleased(){pedalPos.breakPedalUp();}
+
+    public void steeringLeftPressed(){pedalPos.startSteeringLeft();}
+
+    public void steeringRightPressed(){pedalPos.startSteeringRight();}
+
+    public void steeringReleased(){pedalPos.steeringWheelReleased();}
+
+
+    private ACC accManager = new ACC();
```
Comment (AbyssPrince) [inline src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/KeyProcesser.java:86]: Usually, the class is more readable if all fields are on top of the class, followed by the constructor and then getters/setters and other functions.

File: src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/PedalPosition.java [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+                counter++;
+                Sleep();
+            }
+        });
+        valueChangeThread.start();
+    }
+
+    public void gasPedalUp(){
+        gasPedalSwitch = false;
+        gasPedalValueDecrease();;
```
Comment (AbyssPrince) [inline src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/PedalPosition.java:126]: Remove the double `;`

File: src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/PedalPosition.java [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+            return 10;
+        }
+        else{
+            return 15;
+        }
+    }
+
+    private void Sleep(){
+        try {
+            Thread.sleep(100);
```
Comment (AbyssPrince) [inline src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/PedalPosition.java:218]: - Name all functions with lower case, like `sleep()`.
- Why sleep 100 milliseconds? Packets are evaluated through the function bus loop always. If you change the value in a parallel way, while other parts are reading the packet, that might just lead to unnecessary complexity and bugs. So I'm asking, that are these threads necessary to work?

File: src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/Shitfer.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
+package hu.oe.nik.szfmv.automatedcar.systemcomponents;
+
+import hu.oe.nik.szfmv.automatedcar.virtualfunctionbus.VirtualFunctionBus;
+
+public class Shitfer {
+    public enum ShiftPos {P, R, N, D}
+
+    private int acttuallyValue = 0;
```
Comment (AbyssPrince) [inline src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/Shitfer.java:8]: Actually is one `t`.

File: src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/Shitfer.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
+package hu.oe.nik.szfmv.automatedcar.systemcomponents;
+
+import hu.oe.nik.szfmv.automatedcar.virtualfunctionbus.VirtualFunctionBus;
+
+public class Shitfer {
+    public enum ShiftPos {P, R, N, D}
```
Comment (AbyssPrince) [inline src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/Shitfer.java:6]: There is an `enum` for this already on `master` for the `PowerTrain`. I'm fine with this, just saying that it is redundant, possibly should use only one of them in the future.

File: src/main/java/hu/oe/nik/szfmv/automatedcar/virtualfunctionbus/packets/HMIOutputPackets/BaseInputPacket.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
+package hu.oe.nik.szfmv.automatedcar.virtualfunctionbus.packets.HMIOutputPackets;
+
+import hu.oe.nik.szfmv.automatedcar.systemcomponents.Shitfer;
+
+public interface BaseInputPacket {
+
+    int getGasPedalValue();
+    int getBreakPedalValue();
+    int getSteeringWheelValue();
+    int getIndexValue(/*negatív -> balra, pozitív -> jobbra, 0 -> kikapcsolva */);
```
Comment (AbyssPrince) [inline src/main/java/hu/oe/nik/szfmv/automatedcar/virtualfunctionbus/packets/HMIOutputPackets/BaseInputPacket.java:10]: Associating numbers with states are really C'ish. Should use an `enum` instead in these cases.

File: src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/KeyProcesser.java [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+    private void setDebugMode(){
+        if(zeroIsPressed && controlIsPressed){
+            debugMode=!debugMode;
+        }
+        virtualFunctionBus.guiInputPacket.setDebugSwitch(debugMode);
+        virtualFunctionBus.debugModePacket.setDebuggingState(debugMode);
+    }
+
+    private boolean HelpMenuSwitch = false;
```
Comment (AbyssPrince) [inline src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/KeyProcesser.java:52]: Name this field lower-case too: `helpMenuSwitch`

File: src/main/java/hu/oe/nik/szfmv/automatedcar/AutomatedCar.java [hunk 10]
```diff
@@ -38,21 +38,8 @@ public PowerTrain getPowerTrain() {
...
-            case 2:
-                y += 5;
-                break;
-            case 3:
-                x -= 5;
-                break;
-            default:
-                break;
-        }
+        y -= virtualFunctionBus.toPowerTrainPacket.getGasPedalValue()/10;
```
Comment (pintergreg) [inline src/main/java/hu/oe/nik/szfmv/automatedcar/AutomatedCar.java:41]: Kódformázási előírás nincs betartva.

Comment (chowder23) [reply src/main/java/hu/oe/nik/szfmv/automatedcar/AutomatedCar.java:41]: Beállítottam és lefuttattam a fájlokon a Save Actions (ütettem pár entert majd rámentettem, hogy lefusson).  Kijavítottam sok formázási hibát is,  a maradék túlnyomó része Magic Number az elemek elhelyezésénél a Dashboardon.

**Comment by chowder23** [comment]
@pintergreg  Be állítottuk és lefuttatuk a Save Actions.

**Pull request merged**